lib/auth/join_test.go

```diff
@@ -247,6 +249,7 @@ func TestAuth_RegisterUsingToken(t *testing.T) {
 				tc.clock = clockwork.NewRealClock()
 			}
 			a.SetClock(tc.clock)
+			time.Sleep(tc.delay)
```

Comment:
:nail_care: 
I think a `require.Eventually` would be a better fit here.
We could wait for the token to be removed before moving on to the next test

Reply:
Yep, I thought about `require.Eventually` at first but it required a bit more changes and looked less nice. But I changed it to use `require.Eventually` now 👍